apps/dashboard/components/batch-payments/index.tsx

```diff
@@ -0,0 +1,269 @@
+    }
+
+    setProcessPaymentLoading(false)
+  }
+
+  return (
+    <div>
+      <Modal
+        open={modalDetails.open}
+        onClose={() =>
+          setModalDetails({
+            open: false,
+            heading: "",
+            message: "",
+          })
+        }
+        sx={{ display: "flex", justifyContent: "center", alignItems: "center" }}
+      >
+        <Sheet
+          variant="outlined"
+          sx={{
+            maxWidth: 400,
+            borderRadius: "md",
+            p: 3,
+            boxShadow: "lg",
+            display: "flex",
+            flexDirection: "column",
+            gap: 2,
+            alignItems: "left",
+            width: "100%",
+          }}
+        >
+          <ModalClose variant="plain" />
+          <Typography
+            component="h2"
+            id="modal-callback"
+            level="h4"
+            textColor="inherit"
+            fontWeight="lg"
+            mb={1}
+          >
+            {modalDetails.heading}
+          </Typography>
+          <p>{modalDetails.message}</p>
+        </Sheet>
+      </Modal>
+
+      {csvData.length > 0 && paymentDetails && paymentDetails.walletDetails ? (
+        <>
+          <Box
+            sx={{
+              marginBottom: "1em",
+              display: "flex",
+              flexDirection: "row",
+              width: "100%",
+              gap: "1em",
+            }}
+          >
+            <DetailsCard>
+              <Details
+                label="Total Amount"
+                value={`${String(paymentDetails.totalAmount)} ${getCurrencyFromWalletType(
+                  paymentDetails.walletType,
+                )}`}
+              />
+              <Details label="Wallet Type" value={paymentDetails.walletType} />
+              <Details label="Wallet Id" value={paymentDetails.walletDetails.id} />
+              <Details
+                label="Wallet Balance"
+                value={`${
+                  paymentDetails.walletDetails.balance
+                } ${getCurrencyFromWalletType(paymentDetails.walletType)}`}
+              />
+              <Button onClick={processPayments} loading={processPaymentLoading}>
+                Confirm Payment
+              </Button>
+            </DetailsCard>
+            <FileUpload
+              processCsvLoading={processCsvLoading}
+              file={file}
+              setFile={setFile}
+            />
+          </Box>
+          <BatchPaymentsList
+            processedList={csvData}
+            walletType={paymentDetails.walletType}
+          ></BatchPaymentsList>
+        </>
+      ) : (
+        <>
+          <FileUpload
+            processCsvLoading={processCsvLoading}
+            file={file}
+            setFile={setFile}
+          />
+          <Card
+            sx={{
+              margin: "1em auto",
+              padding: "1em",
+            }}
+          >
+            {`Please upload a CSV file containing the following columns: "username" (the
```

Comment:
instead of this text, it would be easier to give the user an empty csv with already the right expected column